sdk/schemaregistry/azure-data-schemaregistry-avro/src/main/java/com/azure/data/schemaregistry/avro/AvroByteDecoder.java

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.data.schemaregistry.avro;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.data.schemaregistry.ByteDecoder;
+import com.azure.data.schemaregistry.SerializationException;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.io.DatumReader;
+import org.apache.avro.io.DecoderFactory;
+import org.apache.avro.specific.SpecificDatumReader;
+
+import java.io.IOException;
+
+/**
+ * ByteDecoder implementation with all Avro-specific functionality required to deserialize byte arrays
+ * given an Avro schema.
+ */
+public class AvroByteDecoder extends AvroCodec
```

Comment:
Name this class `ApacheAvroByteDecoder` so that  we can support Jackson avro later if required. (If this is going away after switching to azure-core implementation, then you can leave it as-is).

Reply:
Leaving as is for now.  I think the plan will be to allow passing in the Jackson serializer, but will default to an Apache implementation.  Dependency story is okay because Jackson has a dep on Apache anyway.